### ISSUE TITLE
opensc: build breaks without OPENSSL_WITH_DEPRECATED

### ISSUE THREAD

**Issue by nwf** [openwrt/packages#4764]
Maintainer: @dangowrt 
Environment: LEDE HEAD

Description:

Attempting to leave out the OPENSSL deprecated APIs results in the following breakage.  It's possible this would be fixed by a version bump to 0.17; haven't tried.

```
pkcs15-prkey.c: In function 'sc_pkcs15_convert_bignum':
pkcs15-prkey.c:614:13: warning: implicit declaration of function 'BN_num_bytes' [-Wimplicit-function-declaration]
  dst->len = BN_num_bytes(bn);
             ^
pkcs15-prkey.c:616:2: warning: implicit declaration of function 'BN_bn2bin' [-Wimplicit-function-declaration]
  BN_bn2bin(bn, dst->data);
  ^
pkcs15-prkey.c: In function 'sc_pkcs15_convert_prkey':
pkcs15-prkey.c:635:51: error: dereferencing pointer to incomplete type 'RSA {aka struct rsa_st}'
   if (!sc_pkcs15_convert_bignum(&dst->modulus, src->n)
                                                   ^
pkcs15-prkey.c:646:3: warning: implicit declaration of function 'RSA_free' [-Wimplicit-function-declaration]
   RSA_free(src);
   ^
pkcs15-prkey.c:654:42: error: dereferencing pointer to incomplete type 'DSA {aka struct dsa_st}'
   sc_pkcs15_convert_bignum(&dst->pub, src->pub_key);
                                          ^
pkcs15-prkey.c:659:3: warning: implicit declaration of function 'DSA_free' [-Wimplicit-function-declaration]
   DSA_free(src);
   ^
Makefile:717: recipe for target 'pkcs15-prkey.lo' failed
```

**Comment by nwf** [comment]
A bump to 0.17 seems to help, but please wait until OpenSC/OpenSC#1141 is resolved before cutting a new release, as that issue means things don't work on big-endian systems including my MIPS gateway.

**Comment by nwf** [comment]
Whoop, forgot to follow up that OpenSC/OpenSC#1141 is now resolved and so the OpenWRT OpenSC package should probably jump up to 0.17. :)

**Comment by nwf** [comment]
This appears to be fixed by 347c563c321fe0693b821b95bf09b66fbbcb77f6 .
